src/platform/PASEUtils.h

```diff
@@ -0,0 +1,45 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <ble/BleConfig.h>
+#include <lib/support/BitFlags.h>
+using namespace chip;
+
+class PASEStatus
+{
+private:
+    static PASEStatus * instance;
+    PASEStatus();
+
+public:
+    enum class PASEFlags : uint8_t
+    {
+        kPASESessionNone      = 0x00,
+        kPASESessionEnabled   = 0x01,
+        kPASESessionFailed    = 0x02,
+        kPASESessionCompleted = 0x03,
+    };
+
+    BitFlags<PASEFlags> pFlags;
```

Comment:
It seems like it would be better to expose state as an enum instead of this bitfield.  For instance, it does not make sense for pFlags to have both kPASESessionNone and kPASESessionEnabled set.

Reply:
Removed.